config/config_test.go

```diff
@@ -0,0 +1,133 @@
+package config
+
+import (
+	"testing"
+	. "github.com/smartystreets/goconvey/convey"
+	"errors"
+	"time"
+)
+
+type processConfigMock struct {
+	prefixCalls []string
+	specCalls   []interface{}
+	err         error
+}
+
+func (m *processConfigMock) Process(prefix string, spec interface{}) error {
+	m.prefixCalls = append(m.prefixCalls, prefix)
+	m.specCalls = append(m.specCalls, spec)
+	return m.err
+}
+
+func newProcessCgfMock() *processConfigMock {
```

Comment:
GO convention for naming Methods, functions or variable names for that matter containing an acronym (e.g. `Cgf`) should be uppercase, like so `newProcessCFGMock()`.

Here I would not bother shortening the word Config to CFG, instead explicitly naming the function as `newProcessConfigMock`.